File: examples/multilanguage/lua/aspect_fit.lua
-- Lua + LÖVE: scale the game canvas proportionally and center it.
local M = {}
M.logicalWidth = 960
M.logicalHeight = 720

function M.viewport(screenWidth, screenHeight)
  local scale = math.min(screenWidth / M.logicalWidth, screenHeight / M.logicalHeight)
  local width = math.floor(M.logicalWidth * scale + 0.5)
  local height = math.floor(M.logicalHeight * scale + 0.5)
  return { x = math.floor((screenWidth - width) / 2), y = math.floor((screenHeight - height) / 2), width = width, height = height, scale = scale }
end

function M.present(canvas)
  local v = M.viewport(love.graphics.getWidth(), love.graphics.getHeight())
  love.graphics.clear(0, 4 / 255, 18 / 255, 1)
  love.graphics.draw(canvas, v.x, v.y, 0, v.scale, v.scale)
end

return M

File: examples/multilanguage/rust/aspect_fit.rs
// Rust: framework-neutral aspect-fit helper for Bevy/macroquad/ggez/SDL2 bindings.
pub const LOGICAL_WIDTH: f32 = 960.0;
pub const LOGICAL_HEIGHT: f32 = 720.0;

#[derive(Debug, Clone, Copy)]
pub struct Viewport { pub x: f32, pub y: f32, pub width: f32, pub height: f32 }

pub fn aspect_fit_viewport(screen_width: f32, screen_height: f32) -> Viewport {
    let scale = (screen_width / LOGICAL_WIDTH).min(screen_height / LOGICAL_HEIGHT);
    let width = LOGICAL_WIDTH * scale;
    let height = LOGICAL_HEIGHT * scale;
    Viewport { x: (screen_width - width) * 0.5, y: (screen_height - height) * 0.5, width, height }
}
